test/e2e/adminactions_redeployvm.go

```diff
@@ -0,0 +1,83 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func pickFirstWorkerVM(ctx context.Context, resourceGroup string) (*mgmtcompute.VirtualMachine, error) {
+	vms, err := Clients.VirtualMachines.List(ctx, resourceGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterate over and return first worker VM that is Succeeded provisioning
+	for _, v := range vms {
+		if strings.Contains(*v.Name, "-worker-") && *v.ProvisioningState == "Succeeded" {
+			return &v, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func pollUntilProvisioningState(ctx context.Context, resourceGroup string, vmName string, desiredState string, timeout time.Duration) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	var err error
+
+	err = wait.PollImmediateUntil(5*time.Second, func() (bool, error) {
+		vm, err := Clients.VirtualMachines.Get(ctx, resourceGroup, vmName, mgmtcompute.InstanceView)
+		if err != nil {
+			return true, err
+		}
+
+		if *vm.ProvisioningState == desiredState {
+			return true, nil
+		}
+
+		return false, nil
+	}, timeoutCtx.Done())
+
+	return err
```

Comment:
Can just return without assigning the variable `err`

```suggestion
	return wait.PollImmediateUntil(5*time.Second, func() (bool, error) {
		vm, err := Clients.VirtualMachines.Get(ctx, resourceGroup, vmName, mgmtcompute.InstanceView)
		if err != nil {
			return true, err
		}

		if *vm.ProvisioningState == desiredState {
			return true, nil
		}

		return false, nil
	}, timeoutCtx.Done())
```